### ISSUE TITLE
No spaces between property access

### ISSUE THREAD

**Issue by c-niederer** [palantir/tslint#4565]
### Rule Suggestion
I suggest a rule which disallows spaces or tabs between property accesses. Newlines would still be allowed.
I imagine a similar behaviour to the spaces before and after arithmetic symbols.

**Is your rule for a general problem or is it specific to your development style?**
General problem - `.` and `,` look very similar so a bug might easily be overlooked.

**What does your suggested rule do?**
Inform the user about the whitespace.

**List several examples where your rule could be used**
_Example1_
```javascript
sampleFunction(some.property. another.property)
```
vs
```javascript
sampleFunction(some.property, another.property)
```
_Example2_
```javascript
const a, b. c = something
```
vs
```javascript
const a, b, c = something
```
_Example3_
```javascript
sampleFunction(some
       .property, something.else)
```
_Example4_
```javascript
sampleFunction(some.
       property, something.else)
```
In example 1 & 2 the user would be notified about the space between the property accesses.
In example 3 & 4 the rule would not apply as newline between property access would be allowed.

**Additional context**
This rule is intended to warn the user about a potential access to undefined during runtime.

**Comment by JoshuaKGoldberg** [comment]
Hi @c-niederer, thanks for the suggestion! Per #4534, TSLint is avoiding adding new code formatting rules to core. If you want to enforce this specific check, it can live happily in a custom rule set outside this repo.
